core/src/test/scala/doric/sem/TransformOpsSpec.scala

```diff
@@ -136,5 +137,58 @@ class TransformOpsSpec
         (11, 1)
       )
     }
+
+    it("drops a single column") {
+      import spark.implicits._
+
+      val df = List(("a", "b")).toDF("col1", "col2")
+
+      val res    = df.drop(colString("col1")).collect().toList
+      val actual = List(Row("b"))
+
+      res shouldBe actual
+    }
+
+    it("drop throws an error when column is not found") {
+      import spark.implicits._
+
+      val df = List(("a", "b")).toDF("col1", "col2")
+
+      intercept[DoricMultiError] {
+        df.drop(colString("invalid")).collect().toList
+      } should containAllErrors(
+        ColumnNotFound("invalid", List("col1", "col2"))
+      )
+    }
+
+    it("drops multiple columns") {
+      import spark.implicits._
+
+      val df = List(("a", "b", 1, 2.0)).toDF("col1", "col2", "col3", "col4")
+
+      val res = df
+        .drop(colString("col2"), colInt("col3"), colDouble("col4"))
+        .collect()
+        .toList
+      val actual = List(Row("a"))
+
+      res shouldBe actual
+    }
+
+    it("drop throws an error when columns are not found") {
+      import spark.implicits._
+
+      val df = List(("a", "b", 1, 2.0)).toDF("col1", "col2", "col3", "col4")
+
+      intercept[DoricMultiError] {
```

Comment:
I know it will catch the error but can you add to this example a column that exists but is not of the expected type?

Reply:
Apologies for the delay on this, it's been a little bit hectic for me the past few weeks. I've added additional tests in [7e98f52](https://github.com/hablapps/doric/pull/366/commits/7e98f521ea97839dbf145935befc32557f35ef5e)